File: app/schemas/product.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class ProductBase(BaseModel):
    title: str
    description: Optional[str] = None
    brand: Optional[str] = None
    sku: str
    price: float
    category_id: int
    status: str = "ACTIVE"


class ProductCreate(ProductBase):
    pass


class ProductUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    brand: Optional[str] = None
    sku: Optional[str] = None
    price: Optional[float] = None
    category_id: Optional[int] = None
    status: Optional[str] = None


class ProductResponse(ProductBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/product.ts
export interface Product {
  id: number;
  title: string;
  sku: string;
  price: number;
  category_id: number;
  description?: string | null;
  brand?: string | null;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface ProductCreate {
  title: string;
  sku: string;
  price: number;
  category_id: number;
  description?: string | null;
  brand?: string | null;
  status?: string;
}

export interface ProductUpdate {
  title?: string;
  sku?: string;
  price?: number;
  category_id?: number;
  description?: string | null;
  brand?: string | null;
  status?: string;
}
